File: back/pola/src/main/java/com/jinjinjara/pola/rag/dto/response/RagSearchResponse.java
package com.jinjinjara.pola.rag.dto.response;

import com.jinjinjara.pola.rag.dto.common.RagSearchSource;

import java.util.List;

public record RagSearchResponse(
        String answer,                 // LLM이 생성한 요약 or 설명
        List<RagSearchSource> sources  // 벡터 검색된 근거 문서들
) {}

File: mobile/Pola/app/src/main/java/com/jinjinjara/pola/data/remote/dto/response/RagSearchResponse.kt
package com.jinjinjara.pola.data.remote.dto.response

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
data class RagSearchResponse(
    @Json(name = "data")
    val data: RagSearchData?,

    @Json(name = "message")
    val message: String,

    @Json(name = "status")
    val status: String,

    @Json(name = "code")
    val code: String? = null
)

@JsonClass(generateAdapter = true)
data class RagSearchData(
    @Json(name = "answer")
    val answer: String,

    @Json(name = "sources")
    val sources: List<SourceDto>
)

@JsonClass(generateAdapter = true)
data class SourceDto(
    @Json(name = "id")
    val id: Long,

    @Json(name = "tags")
    val tags: List<String> = emptyList(),

    @Json(name = "src")
    val src: String,

    @Json(name = "context")
    val context: String,

    @Json(name = "relevanceScore")
    val relevanceScore: Double
)
